lib/tokenize.cpp

```diff
@@ -9414,11 +9416,18 @@ void Tokenizer::simplifyCPPAttribute()
                 if (head && head->str() == "(" && isFunctionHead(head, "{|;")) {
                     head->previous()->isAttributeNodiscard(true);
                 }
-            } else if (Token::findsimplematch(tok->tokAt(2), "maybe_unused", tok->link())) {
+            } else if ((Token::findsimplematch(tok->tokAt(2), "maybe_unused", tok->link()) && hasMaybeUnused)
+                       || (Token::findsimplematch(tok->tokAt(2), "__maybe_unused__", tok->link()) && hasMaybeUnusedUnderscores)) {
                 Token* head = skipCPPOrAlignAttribute(tok)->next();
                 while (isCPPAttribute(head) || isAlignAttribute(head))
                     head = skipCPPOrAlignAttribute(head)->next();
                 head->isAttributeMaybeUnused(true);
+            } else if (Token::findsimplematch(tok->tokAt(2), "gnu :: unused", tok->link()) ||
```

Comment:
hmm.. this `Token::findsimplematch(tok->tokAt(2), "gnu :: unused", tok->link())` is technically redundant. Because if this is true then the function call below will always be true also.